### DISCUSSION TITLE
Apakah AVFoundation perlu Menggunakan UIKit

### DISCUSSION THREAD

**Discussion by alizaenazet** [jerrytjahyono/SelfFit#2] [General]
Untuk akses penggunaan kamera dan audio pada perangkat, menggunakan AVFoundation dengan swfitui apakah masih perlu menggunakan UIKIT ?, sebab didapati di [video dokumentasi masih menggunakan UIKit](https://developer.apple.com/videos/play/wwdc2016/511/) pada view controllernya, begitu pula dibeberapa artikel pun begitu walaupun dominasi menggunakan SwiftUi tetap saja beberapa masih melibatkan penggunaan UiKit pada implementasi AVFoundation-nya.

Begitupula saat sekilas membaca [dokumentasi AVFoundation dan beberapa contoh implementasi disana](https://developer.apple.com/documentation/avfoundation/capture_setup/avcam_building_a_camera_app) didapati penggunaan UIKit. 

```swift
import UIKit
import AVFoundation
import CoreLocation
import Photos

class CameraViewController: UIViewController, AVCaptureFileOutputRecordingDelegate, AVCapturePhotoOutputReadinessCoordinatorDelegate {
	
	let locationManager = CLLocationManager()
    
    // MARK: View Controller Life Cycle
    
    override func viewDidLoad() {
        super.viewDidLoad()

```

**Apakah itu berarti kita mesti belajar UIkit terlebih dahulu sebelum menggunakan AVFoundation ?**

**Comment by TimothyKristanto** [comment]
AVFoundation tidak harus menggunakan UIKit, bisa dengan SwiftUI, cuman dokumentasinya sepeertinya agak jarang

**Reply by alizaenazet** [reply]
tapi kalau semisal kita cuman dapet informasi penggunaan **AVFondation** yang implement **uikit** apa perlu kita belajar **Uikit** juga buat bisa ngeimplement diswift ui aja ?.

**Reply by TimothyKristanto** [reply]
Enda perlu sampai belajar uikit juga si sebenernya, lebih ke arah lihat dokumentasinya AVFoundation saja

**Reply by alizaenazet** [reply]
okeeei, biar kita coba 🫡😂

**Reply by TimothyKristanto** [reply]
WKWKWKKWK, kalo ada resource yang aku temuin tak kirim ke LINE yah
